lib/db.py

```diff
@@ -23,43 +22,51 @@ def __init__(self):
             # force domain socket connection by not supplying host
             # pylint: disable=consider-using-f-string
             if config['postgresql']['host'] is None:
-                self._conn = psycopg.connect("user=%s dbname=%s password=%s port=%s" % (
-                    config['postgresql']['user'],
-                    config['postgresql']['dbname'],
-                    config['postgresql']['password'],
-                    config['postgresql']['port']))
+                self._pool = ConnectionPool(
+                    "postgresql://%s:%s@localhost:%s/%s" % (
+                        config['postgresql']['user'],
+                        config['postgresql']['password'],
+                        config['postgresql']['port'],
+                        config['postgresql']['dbname'],
+                    ),
+                    min_size=1,
+                    max_size=2,
+                    open=True
+                )
             else:
-                self._conn = psycopg.connect("host=%s user=%s dbname=%s password=%s port=%s" % (
-                    config['postgresql']['host'],
-                    config['postgresql']['user'],
-                    config['postgresql']['dbname'],
-                    config['postgresql']['password'],
-                    config['postgresql']['port']))
+                self._pool = ConnectionPool(
+                    "postgresql://%s:%s@%s:%s/%s" % (
+                        config['postgresql']['user'],
+                        config['postgresql']['password'],
+                        config['postgresql']['host'],
+                        config['postgresql']['port'],
+                        config['postgresql']['dbname'],
+                    ),
+                    min_size=1,
+                    max_size=2,
+                    open=True
+                )
```

Comment:
Maybe we should think about implementing a 'check' function?

https://www.psycopg.org/psycopg3/docs/api/pool.html#psycopg_pool.ConnectionPool